modules/common/src/bugsnag/bugsnag.filter.ts

```diff
@@ -0,0 +1,44 @@
+import {Catch, ExceptionFilter, HttpStatus} from '@nestjs/common';
+import {ArgumentsHost} from '@nestjs/common/interfaces/features/arguments-host.interface';
+import * as httpContext from 'express-http-context';
+import * as jsonStringifySafe from 'json-stringify-safe';
+import {REQUEST_UNIQUE_ID_KEY} from '../constants';
+import {clearBreadcrumbs, getBreadcrumbs} from './breadcrumbs';
+import {BugsnagClient} from './bugsnag.client';
+import {BugsnagSeverity} from './interfaces';
+
+@Catch()
+export class BugsnagErrorFilter implements ExceptionFilter {
+    constructor (
+        private readonly bugsnagClient: BugsnagClient,
+        private readonly loggerMethod?: Function,
+    ) {}
+
+    public catch (err: any, host: ArgumentsHost): any {
```

Comment:
I can't be 100% sure what we get here, because it will receive whatever is thrown at the controller... I think that is `HttpException` most of the time, but if someone were to throw a string, that's what we would get here... 

I will try to make this a bit safer though, and validate the types I'm checking.